proposals/NNNN-package-manager-product-definitions.md

```diff
@@ -7,135 +7,123 @@
 
-The products represent the publicly vended package outputs on which any client package can depend, and also define the characteristics of the produced artifacts.
+The products represent the publicly vended package outputs on which any client package can depend.  Other artifacts might also be created as part of building the package, but what the products specifically define are the conceptual "outputs" of the package, i.e. those that make sense to think of as something a client package can depend on and use.  Examples of artifacts that are not necessarily products include built unit tests and helper tools that are used only during the build of the package.
 
-As mentioned earlier, Swift Package Manager already infers a set of implicit products based on the set of targets in the package.  One open question is whether inference rules will be supported and documented as part of this feature, in order to allow existing packages to continue to work unmodified and to avoid the need for explicit product definitions for simple packages.  While the inference rules will need to be supported for existing packages, it is still an open question whether product will be inferred for packages using the Swift Pacakge Manager 4.0 version of the API.
-
-The inference rules that apply to legacy packages are detailed in the next section of this document.
-
-Even if Swift Pacakge Manager 4.0 does infer products for non-legacy packages, products will only be inferred if there are no explicit product definitions in the package.  The reason is that the interaction of a mixture of implicit and explicit products is likely to lead to significant complexity and to be a source of both confusion and bugs.
-
-An example of a package containing explicit product definitions:
+An example of a package that defines two library products and one executable product:
 
 ```
 let package = Package(
-    name: "Hello",
+    name: "MyServer",
     targets: [
-        Target(name: "Foo"),
-        Target(name: "Bar"),
-        Target(name: "Baz", dependencies: ["Foo", "Bar"]),
-        Target(name: "Exe", dependencies: ["Foo"])
+        Target(name: "Utils"),
+        Target(name: "HTTP", dependencies: ["Utils"]),
+        Target(name: "ClientAPI", dependencies: ["HTTP", "Utils"]),
+        Target(name: "ServerAPI", dependencies: ["HTTP"]),
+        Target(name: "ServerDaemon", dependencies: ["ServerAPI"])
     ],
     products: [
-        Library(name: "Lib1", type: .static, targets: ["Bar"]),
-        Library(name: "Lib2", type: .dynamic, targets: ["Baz"]),
-        Executable(name: "Exe", targets: ["Exe"]),
+        Library(name: "ClientLib", type: .static, targets: ["ClientAPI"]),
+        Library(name: "ServerLib", type: .dynamic, targets: ["ServerAPI"]),
+        Executable(name: "myserver", targets: ["ServerDaemon"]),
```

Comment:
It was at a global scope.  But in my prototype I have now put them as static functions under the `Product` struct, which means that they can be accessed as `.Library` and `.Executable`.  This is similar to how the package dependencies work.